File: backend/src/pomelo_orbit/domain/ci/repositories/artifact.py
"""CI 制品仓储接口"""

from abc import ABC, abstractmethod

from pomelo_orbit.domain.ci.entities import Artifact


class ArtifactRepository(ABC):
    @abstractmethod
    def find_by_run(self, project_id: str, pipeline_run_id: str) -> list[Artifact]: ...

    @abstractmethod
    def find_paginated_by_project_id(
        self,
        project_id: str,
        page: int = 1,
        per_page: int = 20,
        repository_id: str | None = None,
        template_id: str | None = None,
        search: str | None = None,
    ) -> tuple[list[Artifact], int]: ...

    @abstractmethod
    def save(self, artifact: Artifact) -> None: ...

File: internal/application/ci/artifact.go
package cisvc

import (
	"context"
	"strings"

	apperror "gitee.com/leoninew/PomeloOrbit-go/internal/common/errors"
	"gitee.com/leoninew/PomeloOrbit-go/internal/model"
	"gitee.com/leoninew/PomeloOrbit-go/internal/repository"
)

type ArtifactListInput struct {
	ProjectId    string
	RepositoryId string
	TemplateId   string
	Search       string
	Page         int
	PerPage      int
}

func (s Service) ListArtifacts(ctx context.Context, userId string, input ArtifactListInput) (repository.Page[model.Artifact], error) {
	projectId := strings.TrimSpace(input.ProjectId)
	if projectId == "" {
		return repository.Page[model.Artifact]{}, apperror.New(apperror.KindValidation, "project_id is required")
	}
	if err := s.ensureProjectMembership(ctx, projectId, userId); err != nil {
		return repository.Page[model.Artifact]{}, err
	}
	if err := s.ensurePipelineRunRepositoryFilter(ctx, projectId, input.RepositoryId); err != nil {
		return repository.Page[model.Artifact]{}, err
	}
	if err := s.ensurePipelineRunTemplateFilter(ctx, projectId, input.TemplateId); err != nil {
		return repository.Page[model.Artifact]{}, err
	}
	items, err := s.store.ListArtifacts(ctx, projectId, input.RepositoryId, input.TemplateId, input.Page, input.PerPage, input.Search)
	if err != nil {
		return repository.Page[model.Artifact]{}, apperror.Wrap(apperror.KindInternal, "Failed to list artifacts", err)
	}
	return items, nil
}
